### ISSUE TITLE
File headers not showing up in newly created file with csv layout

### ISSUE THREAD

**Issue by GitMarJansen** [NLog/NLog#2086]
Refers to closed issue: https://github.com/NLog/NLog/issues/1921

I also have this issue. I created a test by getting the latest nugetpackage for NLOG (config and xsd) and write a small piece of config and code. My archived log contains the header, the currentlog doesn't.
[NLogCsvIssue.zip](https://github.com/NLog/NLog/files/967933/NLogCsvIssue.zip)

**Comment by snakefoot** [comment]
This is the contents of "csv-log.txt":
```
Date
2017-05-01T16:34:25
```
This is the content for "archive/csv-log-20170501.txt":

```
Date
2017-05-01T16:30:09
Date
2017-05-01T16:31:50
```

I see lots of headers, maybe one too many. Guess you have to add your internal-log.

**Comment by GitMarJansen** [comment]
[logs.zip](https://github.com/NLog/NLog/files/968249/logs.zip)
I tries at home on a differrent machine and get the same result as before. See attached. The logging in the archived files do not show duplicates.
I will run another test with internal logging turned on.

**Comment by GitMarJansen** [comment]
The log:
[nlog-internal.zip](https://github.com/NLog/NLog/files/968269/nlog-internal.zip)

**Comment by snakefoot** [comment]
The log for my result:
[nlog-internal.txt](https://github.com/NLog/NLog/files/968703/nlog-internal.txt)

**Comment by 304NotModified** [comment]
@GitMarJansen what's your NLog version?

**Comment by GitMarJansen** [comment]
I am using the latest version from nuget in combination with vs2017.

Met vriendelijke groet,
Hilmar Jansen

Van: Julian Verdurmen
Verzonden: maandag 1 mei 2017 22:23
Aan: NLog/NLog
CC: Hilmar Jansen; Mention
Onderwerp: Re: [NLog/NLog] File headers not showing up in newly created filewith csv layout (#2086)

@GitMarJansen what's your NLog version?

**Comment by snakefoot** [comment]
@GitMarJansen Could you debug and check why FileTarget.FinalizeFile() doesn't remove the filename from FileTarget.initializedFiles-dictionary. And then also check why FileTarget.InitializeFile() doesn't set writeHeader = true ?

**Comment by GitMarJansen** [comment]
It is not removed because the filename in the list is with forward slashes, while the filename to be removed is with backslashes.

<img width="635" alt="nlog-issue-csv" src="https://cloud.githubusercontent.com/assets/1133052/25606392/5034e044-2f11-11e7-83e1-2f8b5f79fa1d.png">

**Comment by snakefoot** [comment]
Have added a test case with #1993, that tests FileTarget-Header-Layout together with FileTarget-Archive and non-standard-path-layout.

No intention of trying to fiddle with merging archive-files that contains header- and footer-layout. So I guess that feature is still missing. Guess you have to live with using archiveNumbering="DateAndSequence"

**Comment by snakefoot** [comment]
@304NotModified #1993 fixes lots of issues with mismatch in back-slash / for-slash. And any problems found with NLog 4.5 should be created as new issues. Please resolve this one.

**Comment by 304NotModified** [comment]
> @304NotModified #1993 fixes lots of issues with mismatch in back-slash / for-slash. And any problems found with NLog 4.5 should be created as new issues. Please resolve this one.

@snakefoot. I'm sorry, but what should I do now?

**Comment by snakefoot** [comment]
Close this issue as resolved with NLog ver 4.5
________________________________

**Comment by 304NotModified** [comment]
🎉 thanks!
